osm/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/osm/WorkflowOSM.groovy

```diff
@@ -427,34 +425,105 @@ IProcess osm_processing() {
-                                }
-                                def results = geoIndicators.getResults()
-                                results.put("buildingTableName", buildingTableName)
-                                results.put("roadTableName", roadTableName)
-                                results.put("railTableName", railTableName)
-                                results.put("hydrographicTableName", hydrographicTableName)
-                                results.put("vegetationTableName", vegetationTableName)
-                                results.put("imperviousTableName", imperviousTableName)
-                                if (outputFolder && geoIndicatorsComputed && ouputTableFiles) {
-                                    saveOutputFiles(h2gis_datasource, id_zone, results, ouputTableFiles, outputFolder, "osm_", outputSRID, reproject)
+                                //If the config file contains the parameter estimateHeight then
+                                //the indicatorUse must be updated
+
+                                if (estimateHeight) {
+                                    //First pass to compute all indicators that will be used to estimate the height
+                                    IProcess geoIndicators = ProcessingChain.GeoIndicatorsChain.computeAllGeoIndicators()
+                                    if (!geoIndicators.execute(datasource: h2gis_datasource, zoneTable: zoneTableName,
+                                            buildingTable: buildingTableName, roadTable: roadTableName,
+                                            railTable: railTableName, vegetationTable: vegetationTableName,
+                                            hydrographicTable: hydrographicTableName, imperviousTable: imperviousTableName,
+                                            indicatorUse:["URBAN_TYPOLOGY"],
+                                            svfSimplified: true, prefixName: processing_parameters.prefixName,
+                                            mapOfWeights: processing_parameters.mapOfWeights,
+                                            lczRandomForest: processing_parameters.lczRandomForest)) {
+                                        error "Cannot build the geoindicators for the zone $id_zone to estimate the building height"
+                                        geoIndicatorsComputed = false
+                                    } else {
+                                        geoIndicatorsComputed = true
+                                        info "The whole indicators for the ${id_zone} has been processed to estimate the building height"
+                                    }
+                                    //Let's go to select the building's id that must be processed to fix the height
+                                    if(geoIndicatorsComputed){
+                                        info "Extracting the building to estimate their height for the ${id_zone}"
```

Comment:
info "Extracting the building having no height information for the ${id_zone} and estimate it"